frontend/portals/CartCheckoutButton/CartCheckoutButton.jsx

```diff
@@ -19,8 +23,25 @@ const CartCheckoutButton = ({
   }
 
   useEffect(() => {
+    /**
+     * Retry promise to handle maximum number of retries and provide error log
+     * @returns {Function}
+     */
+    const retry = () => new Promise((resolve, reject) => {
```

Comment:
After checking the code I think it won't really work. Let's talk about it offline.